bogenliga/bogenliga-business-logic/src/main/java/de/bogenliga/application/business/competitionclass/impl/business/CompetitionClassComponentImpl.java

```diff
@@ -51,81 +51,63 @@ public CompetitionClassComponentImpl(
 
-        final CompetitionClassBE competitionClassBE = CompetitionClassMapper.toCompetitionClassBE.apply(competitionClassDO);
-
-        int year = Calendar.getInstance().get(Calendar.YEAR);
-        Long alterMin = year - competitionClassDO.getKlasseJahrgangMin();
-        Long alterMax = year - competitionClassDO.getKlasseJahrgangMax();
-
-        competitionClassBE.setKlasseAlterMin(alterMin);
-        competitionClassBE.setKlasseAlterMax(alterMax);
+        final CompetitionClassBE competitionClassBE = convertDOToBE(competitionClassDO);
 
-        final CompetitionClassBE persistedCompetitionClassBE = competitionClassDAO.create(competitionClassBE,
-                currentDsbMitglied);
+        final CompetitionClassBE persistedCompetitionClassBE = competitionClassDAO.create(competitionClassBE, currentDsbMitglied);
 
         return CompetitionClassMapper.toCompetitionClassDO.apply(persistedCompetitionClassBE);
     }
 
-    @Override
-    public CompetitionClassDO update(final CompetitionClassDO competitionClassDO, final long currentDsbMitblied) {
-        checkCompetitionClassDO(competitionClassDO, currentDsbMitblied);
-        Preconditions.checkArgument(competitionClassDO.getId() >= 0, PRECONDITION_MSG_KLASSE_ID);
 
-        final CompetitionClassBE competitionClassBE = CompetitionClassMapper.toCompetitionClassBE.apply(competitionClassDO);
 
-        int year = Calendar.getInstance().get(Calendar.YEAR);
-        Long alterMin = year - competitionClassDO.getKlasseJahrgangMin();
-        Long alterMax = year - competitionClassDO.getKlasseJahrgangMax();
+    @Override
+    public CompetitionClassDO update(final CompetitionClassDO competitionClassDO, final long currentDsbMitglied) {
+        checkCompetitionClassDO(competitionClassDO, currentDsbMitglied);
+        Preconditions.checkArgument(competitionClassDO.getId() >= 0, PRECONDITION_MSG_KLASSE_ID);
 
-        competitionClassBE.setKlasseAlterMin(alterMin);
-        competitionClassBE.setKlasseAlterMax(alterMax);
+        final CompetitionClassBE competitionClassBE = convertDOToBE(competitionClassDO);
 
-        final CompetitionClassBE persistedCompetitionClassBE = competitionClassDAO.update(competitionClassBE,currentDsbMitblied);
+        final CompetitionClassBE persistedCompetitionClassBE = competitionClassDAO.update(competitionClassBE,currentDsbMitglied);
 
         return CompetitionClassMapper.toCompetitionClassDO.apply(persistedCompetitionClassBE);
     }
 
+    // Creates a new CompetitionClassDO Object from a CompetitionClassBE Object
+    private CompetitionClassDO convertBEToDO (CompetitionClassBE competitionClassBE) {
+        int year = Calendar.getInstance().get(Calendar.YEAR);
+        competitionClassBE.setKlasseAlterMin(year - competitionClassBE.getKlasseAlterMin());
+        competitionClassBE.setKlasseAlterMax(year - competitionClassBE.getKlasseAlterMax());
+        return CompetitionClassMapper.toCompetitionClassDO.apply(competitionClassBE);
```

Comment:
Das dann lieber auch in die Mapper Klasse (auch bei convertDOToBE). Dann ist das Mapping auch tatsächlich in der Mapper Klasse und nicht halb in der Business Logik, halb in der Mapper Klasse.

Reply:
Könntest du das noch etwas ausführlicher beschreiben? 

Die Codestelle, die du da markiert hast, hat bereits vor meiner Änderung in der CompetitionClassComponentImpl.java existiert. Ich hab nur den doppelten Code da rausgenommen und in eine Funktion extrahiert. Die Mapping hat davor schon in der Implementierungs-Klasse statt gefunden.

Ich kann natürlich das ganze auch in die Mapper Klasse verschieben.

Reply:
Hab ich gesehen, dass die schon vorher drin war. Aber wir wollen ja ein bisschen wartbarer und besser werden - deshalb bitte ganz in die Mapper Klasse verschieben :-)
Wir werden immer wieder auf Codestellen stoßen, die noch nicht ganz sauber sind & wenn wir die Sachen sowieso schon anfassen, können wir auch statt der 80% Lösung die 100% Lösung implementieren :-)
Grundsätzlich finde ich es aber gut, dass Du versuchst, doppelten Code zu vermeiden und Dich traust, zu refactoren :-)